diff --git a/node_modules/web3-eth/node_modules/web3-utils/.DS_Store b/node_modules/web3-eth/node_modules/web3-utils/.DS_Store
new file mode 100644
index 0000000..5172429
Binary files /dev/null and b/node_modules/web3-eth/node_modules/web3-utils/.DS_Store differ
diff --git a/node_modules/web3-eth/node_modules/web3-utils/src/utils.js b/node_modules/web3-eth/node_modules/web3-utils/src/utils.js
index 379e3e9..62f91d7 100644
--- a/node_modules/web3-eth/node_modules/web3-utils/src/utils.js
+++ b/node_modules/web3-eth/node_modules/web3-utils/src/utils.js
@@ -84,10 +84,10 @@ var toTwosComplement = function (number) {
  */
 var isAddress = function (address) {
     // check if it has the basic requirements of an address
-    if (!/^(0x)?[0-9a-f]{40}$/i.test(address)) {
+    if (!/^(0x)?[0-9a-f]{40}$/i.test(address) && !/^(xdc)?[0-9a-f]{40}$/i.test(address)) {
         return false;
         // If it's ALL lowercase or ALL upppercase
-    } else if (/^(0x|0X)?[0-9a-f]{40}$/.test(address) || /^(0x|0X)?[0-9A-F]{40}$/.test(address)) {
+    } else if (/^(0x|0X|xdc|XDC)?[0-9a-f]{40}$/.test(address) || /^(0x|0X|xdc|XDC)?[0-9A-F]{40}$/.test(address)) {
         return true;
         // Otherwise check each case
     } else {
@@ -106,9 +106,14 @@ var isAddress = function (address) {
  */
 var checkAddressChecksum = function (address) {
     // Check each case
-    address = address.replace(/^0x/i,'');
-    var addressHash = sha3(address.toLowerCase()).replace(/^0x/i,'');
-
+    var addressHash;
+    if (/^(xdc)?[0-9a-f]{40}$/i.test(address)) {
+      address = address.toLowerCase().replace(/^xdc/i, '');
+      addressHash = utils.sha3(address).replace(/^xdc/i, '');
+    } else {
+      address = address.replace(/^0x/i,'');
+      addressHash = sha3(address.toLowerCase()).replace(/^0x/i,'');
+    }
     for (var i = 0; i < 40; i++ ) {
         // the nth letter should be uppercase if the nth digit of casemap is 1
         if ((parseInt(addressHash[i], 16) > 7 && address[i].toUpperCase() !== address[i]) || (parseInt(addressHash[i], 16) <= 7 && address[i].toLowerCase() !== address[i])) {
